service/history/queues/executable.go

```diff
@@ -112,6 +116,7 @@ type (
 		scheduledTime                 time.Time
 		userLatency                   time.Duration
 		lastActiveness                bool
+		resourceExhaustedCount        int
```

Comment:
This is the # of continuous resource exhausted errors. Using this instead of `attempt` so that task won't get a very high penalty after get one resource exhausted error after retrying due to some other reasons.